src/overlays/actors/ovl_En_Sw/z_en_sw.h

```diff
@@ -14,39 +14,48 @@ typedef struct EnSw {
     /* 0x0190 */ EnSwActionFunc actionFunc;
     /* 0x0194 */ ColliderJntSph collider;
     /* 0x01B4 */ ColliderJntSphElement sphs[1];
-    /* 0x01F4 */ Color_RGBA8 unk_1F4;
+    /* 0x01F4 */ Color_RGBA8 limb4Col; // never set past black.
     /* 0x01F8 */ Vec3s jointTable[30];
     /* 0x02AC */ Vec3s morphTable[30];
-    /* 0x0360 */ u8 unk_360;
-    /* 0x0364 */ Vec3f unk_364;
+    /* 0x0360 */ u8 goldHiddenBool; // set when revealed, unset when landing.
+    /* 0x0364 */ Vec3f normalVec;
     /* 0x0370 */ Vec3f unk_370;
     /* 0x037C */ Vec3f unk_37C;
-    /* 0x0388 */ s16 unk_388;
-    /* 0x038A */ s16 unk_38A;
-    /* 0x038C */ s16 unk_38C;
-    /* 0x038E */ s16 unk_38E;
-    /* 0x0390 */ s16 unk_390;
-    /* 0x0392 */ s16 unk_392;
-    /* 0x0394 */ s16 unk_394;
+    /* 0x0388 */ s16 aniTimer;
+    /* 0x038A */ s16 animationSpeed;
+    /* 0x038C */ s16 waitTimer;
+    /* 0x038E */ s16 crawlTimer;
+    /* 0x0390 */ s16 painTimer0;
+    /* 0x0392 */ s16 painTimer1;
+    /* 0x0394 */ s16 deathFlames;
     /* 0x0396 */ char unk_396[0x42];
     /* 0x03D8 */ MtxF unk_3D8;
     /* 0x0418 */ char unk_418[8];
-    /* 0x0420 */ f32 unk_420;
+    /* 0x0420 */ f32 rotateMag;
     /* 0x0424 */ char unk_424[0x8];
     /* 0x042C */ u8 unk_42C;
-    /* 0x0430 */ CollisionPoly* unk_430;
-    /* 0x0434 */ Vec3f unk_434;
-    /* 0x0440 */ s16 unk_440;
-    /* 0x0442 */ s16 unk_442;
-    /* 0x0444 */ s16 unk_444;
-    /* 0x0446 */ s16 unk_446;
-    /* 0x0448 */ Vec3f unk_448;
-    /* 0x0454 */ Vec3f unk_454;
-    /* 0x0460 */ Vec3f unk_460;
-    /* 0x046C */ Vec3f unk_46C;
-    /* 0x0478 */ Vec3f unk_478;
-    /* 0x0484 */ Vec3f unk_484;
+    /* 0x0430 */ CollisionPoly* wallPoly;
+    /* 0x0434 */ Vec3f unk_434; // set during EnSW_LineTestWall, never read.
+    /* 0x0440 */ s16 sfxTimer;
+    /* 0x0442 */ s16 dashTimer;
+    /* 0x0444 */ s16 rotZTarget;
+    /* 0x0448 */ Vec3f targetPos;
+    /* 0x0454 */ Vec3f lineCast0; // used for line tests of walls/player
+    /* 0x0460 */ Vec3f lineCast1;
+    /* 0x046C */ Vec3f lineCast2;
+    /* 0x0478 */ Vec3f lineCast3;
+    /* 0x0484 */ Vec3f walllCast;
     /* 0x0490 */ char unk_490[0x48];
 } EnSw; // size = 0x04D8
 
+#define SW_GOLDTYPE(params) ((params & 0xE000) >> 0xD)
```

Comment:
using `params` as the macro's arg rather than `thisx` appears to be more compatible, as there are 3 different use cases of `thisx->params` `thisx->params - 0x8000` and `this->actor.params`

Reply:
Actors don't always put all their params in `Actor.params` (mostly for "not everything fits there" reasons), so to abstract where the data goes we decided these macros should take a thisx

(for the `this->actor.params` uses it would mean passing `&this->actor` to the macro)

Admittedly `ENSW_GET_GOLDTYPE((thisx->params - 0x8000))` is a weird one, didn't notice that... I would suggest keeping that one not using the macro for now, a comment could still indicate it's the type
The line below it uses its shift anyway

Reply:
It should be noted that the `(thisx)->params` convention breaks in MM (and it's in `en_sw` as well):

https://github.com/zeldaret/mm/blob/master/src/overlays/actors/ovl_En_Sw/z_en_sw.h#L10
https://github.com/zeldaret/mm/blob/master/src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c#L106

There really isn't much reason for us to tie params to an actor instance, yet we do so because we are able to get away with it in most cases. I don't think this issue has ever really been discussed properly.

Reply:
Can you show in code where there is a macro being passed params instead of thisx?

> I don't think this issue has ever really been discussed properly.

It has, Tharo even made https://github.com/zeldaret/oot/pull/1359